cumulusci/tasks/metadeploy.py

```diff
@@ -0,0 +1,123 @@
+import json
+import requests
+
+from cumulusci.core.config import BaseConfig
+from cumulusci.core.exceptions import PlanNotFound
+from cumulusci.core.tasks import BaseTask
+from cumulusci.core.flowrunner import FlowCoordinator
+
+
+class Publish(BaseTask):
+    """Publishes an installation plan to MetaDeploy.
+    """
+
+    task_options = {
+        "plan": {
+            "description": "Name of the installation plan to publish",
+            "required": True,
+        },
+        "tag": {"description": "Name of git tag to publish"},
+    }
+
+    def _init_task(self):
+        metadeploy_service = self.project_config.keychain.get_service("metadeploy")
+        self.base_url = metadeploy_service.url
+        self.api = requests.Session()
+        self.api.headers["Authorization"] = "token {}".format(metadeploy_service.token)
+        self.metadeploy_config = self.project_config.metadeploy
+        if self.metadeploy_config is None:
+            raise PlanNotFound(
+                "No metadeploy configuration found. "
+                "Did you forget to configure it in cumulusci.yml?"
+            )
+        plan_config = getattr(
+            self.project_config, "metadeploy__plans__{}".format(self.options["plan"])
+        )
+        if plan_config is None:
+            raise PlanNotFound("{} plan not found.".format(self.options["plan"]))
+        self.plan_config = BaseConfig(plan_config)
+
+    def _call_api(self, method, url, **kwargs):
```

Comment:
again, only really matters if you have to write more MetaDeploy tasks or a MetaDeploy api client for other purposes.


... but i'd add an arg (default false) that causes this to only get the first page instead of running the full cursor